dfir_lang/src/graph/ops/state_by.rs

```diff
@@ -18,16 +19,22 @@ use crate::diagnostic::{Diagnostic, Level};
 /// use lattices::set_union::{CartesianProductBimorphism, SetUnionHashSet, SetUnionSingletonSet};
 ///
 /// my_state = source_iter(0..3)
-///     -> state_by::<SetUnionHashSet<usize>>(SetUnionSingletonSet::new_from);
+///     -> state_by::<SetUnionHashSet<usize>>(SetUnionSingletonSet::new_from, ::std::default::Default::default);
 /// ```
+/// The 2nd argument into `state_by` is a factory function that can be used to supply a custom
+/// initial value for the backing state. The initial value is still expected to be bottom (and will
+/// be checked). This is useful for doing things like pre-allocating buffers, etc. In the above
+/// example, it is just using `Default::default()`
+///
+///
```

Comment:
Would be nice to have an example of the constructor if you can think of a simple use case

Reply:
Would be good to add a note `state()` is equivalent to `state_by(identity, Default::default)` to both operators